tests/integration/targets/test_awsconfig_apigateway_with_lambda_integration/tasks/main.yml

```diff
@@ -0,0 +1,76 @@
+---
+- name: Test Role 'awsconfig_apigateway_with_lambda_integration'
+  vars:
+    awsconfig_apigateway_with_lambda_integration_lambda_function_file: "{{ role_path }}/files/server.py"
+  block:
+    - name: Create temporary directory for test
+      ansible.builtin.tempfile:
+        suffix: .py
+        state: directory
+      register: __tempdir
+
+    - name: Copy temporary file
+      ansible.builtin.copy:
+        src: server.py
+        dest: "{{ __tempdir.path }}"
+        mode: 0755
+
+    # Test create API gateway
+    - name: Create API Gateway
+      ansible.builtin.include_role:
+        name: cloud.aws_ops.awsconfig_apigateway_with_lambda_integration
+      vars:
+        awsconfig_apigateway_with_lambda_integration_lambda_function_file: "{{ __tempdir.path }}/server.py"
+
+    - name: Ensure the role has defined the invoke url as output
+      ansible.builtin.assert:
+        that:
+          - awsconfig_apigateway_with_lambda_integration__invoke_url is defined
+
+    - name: Calling URL using random path
+      ansible.builtin.uri:
+        url: "{{ awsconfig_apigateway_with_lambda_integration__invoke_url }}/main?resource_prefix={{ resource_prefix }}"
+      register: __uri_result
+
+    - name: Ensure server returns empty result
+      ansible.builtin.assert:
+        that:
+          - __uri_result.json == ""
+
+    - name: Calling URL using ansible-test path
+      ansible.builtin.uri:
+        url: "{{ awsconfig_apigateway_with_lambda_integration__invoke_url }}/ansible-test?resource_prefix={{ resource_prefix }}"
+      register: __uri_result
+
+    - name: Ensure result is as expected
+      ansible.builtin.assert:
+        that:
+          - __uri_result.json == "Running ansible-test with Resource prefix {{ resource_prefix }}"
+
+    # Test: Delete API gateway
+    - name: Delete API Gateway
+      ansible.builtin.include_role:
+        name: cloud.aws_ops.awsconfig_apigateway_with_lambda_integration
+      vars:
+        awsconfig_apigateway_with_lambda_integration_operation: delete
+
+    - name: Validate that URL is not valid after the API Gateway has been deleted
+      ansible.builtin.uri:
+        url: "{{ awsconfig_apigateway_with_lambda_integration__invoke_url }}/ansible-test?resource_prefix={{ resource_prefix }}"
+      ignore_errors: true
+      register: __uri_result
+      failed_when: __uri_result is successful
+
+  rescue:
```

Comment:
Why not use `always` here like we usually do? The use of `rescue` in this case is going to make a failure in the test suite show up as successful in the CI. In fact, if you look at the integration test run for this PR you can see that this exact scenario happened. The tests failed, but the CI shows green.